src/transport/transport.cc

```diff
@@ -731,77 +814,72 @@ void ConvectionTransport::create_transport_matrix_mpi() {
     ElementAccessor<3> el2;
     ElementAccessor<3> elm;
     const Edge *edg;
-    int j, np, rank;
+    int j;
     LongIdx new_j, new_i;
     double aij, aii;
         
     MatZeroEntries(tm);
 
-    MPI_Comm_rank(PETSC_COMM_WORLD, &rank);
-    MPI_Comm_size(PETSC_COMM_WORLD, &np);
-
     double flux, flux2, edg_flux;
 
     aii = 0.0;
 
-    for (unsigned int loc_el = 0; loc_el < el_ds->lsize(); loc_el++) {
-        elm = mesh_->element_accessor( el_4_loc[loc_el] );
-        new_i = row_4_el[ elm.idx() ];
-
-        for (unsigned int si=0; si<elm->n_sides(); si++) {
-            // same dim
-            flux = mh_dh->side_flux( *(elm.side(si)) );
-            if (elm.side(si)->cond() == NULL) {
-                 edg = elm.side(si)->edge();
-                 edg_flux = 0;
-                 for( int s=0; s < edg->n_sides; s++) {
-                   flux2 = mh_dh->side_flux( *(edg->side(s)) );
-                   if ( flux2 > 0)  edg_flux+= flux2;
-                 }
-                 for(unsigned int s=0; s<edg->n_sides; s++)
-                    // this test should also eliminate sides facing to lower dim. elements in comp. neighboring
-                    // These edges on these sides should have just one side
-                    if (edg->side(s) != elm.side(si)) {
-                        j = edg->side(s)->element().idx();
+    unsigned int loc_el = 0;
+    for ( DHCellAccessor dh_cell : dh_->own_range() ) {
+        new_i = row_4_el[ dh_cell.elm_idx() ];
+        elm = dh_cell.elm();
+        for( DHCellSide cell_side : dh_cell.side_range() ) {
+            flux = this->side_flux(elm, cell_side.side()->side_idx());
```

Comment:
Different of fluxes in master and new code - for test with two 2D elements.
Values in master: 
0.00277778 0.0194444 -0.0222222 0.0222222 -0.0277778 0.00555556
Values in this branch:
0.002777780.00277778 -0.0157135 0.0157135 -0.0277778 0.00555556

Values of matrix tm at the end of method in master:
Mat Object: 1 MPI processes
  type: seqaij
row 0: (0, -0.0444444)  (1, 0.0444444)
row 1: (0, 0.)  (1, -0.0555556)

Matrix in this branch:
Mat Object: 1 MPI processes
  type: seqaij
row 0: (0, -0.0444444)  (1, 0.031427)
row 1: (0, 0.)  (1, -0.0425381)